### PR TITLE
[JENKINS-64483] Hide collapse link if it isn't possible

### PR DESCRIPTION
**Pull request by daniel-beck**
Also document the optional 'collapseText'

See [JENKINS-64483](https://issues.jenkins-ci.org/browse/JENKINS-64483).

Follows up on https://github.com/jenkinsci/jenkins/pull/1014

### Proposed changelog entries

* Hide collapse icon in sidepanel widgets if they cannot be collapsed

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/lib/layout/pane.jelly
Comment (res0nance) [inline core/src/main/resources/lib/layout/pane.jelly:50]: I can't seem to see this being used anywhere

Comment (timja) [reply core/src/main/resources/lib/layout/pane.jelly:50]: line 77

**Review by timja — COMMENTED** [review]
sounds ok, I don't know how to test this out though, do you have an example of any widgets?

**Comment by daniel-beck** [comment]
> sounds ok, I don't know how to test this out though, do you have an example of any widgets?

No; just something I noticed when trying to reproduce an issue mentioned elsewhere.

It's trivial to build such a plugin (`@Extension public class Whatever extends Widget` plus a `index.jelly` with a basic `f:pane`). Alternatively, patch the core widgets (e.g. https://github.com/jenkinsci/jenkins/blob/7a5d49594c988c9145ad087b591e8ca5c2f5cca5/core/src/main/resources/lib/hudson/executors.jelly#L137 ) to not pass ID and see what happens.

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
LGTM, thanks for cleaning it up

**Pull request merged**
